patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,7 +9,27 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
-@@ -181,7 +181,7 @@
+                         break;
+                     }
+ 
+-                    tileentitybeacon$beamsegment.func_177262_a();
+-                    continue;
++                    // Allow any block to change the beacon beam color
++                    float[] color = iblockstate.func_177230_c().getBeaconColor(field_145850_b, iblockstate);
++                    if(color == null){
++                        tileentitybeacon$beamsegment.func_177262_a();
++                        continue;
++                    }
```

Comment:
oh yes, that's bad. your patch takes vanilla code and moves it into that if statement. don't move the vanilla code, structure the patch around it